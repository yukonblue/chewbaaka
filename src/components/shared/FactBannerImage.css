/**
 * FactBannerImage.css
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Aug 11, 2020
 * Updated  : Aug 23, 2021
 */

img.FactBannerImage {
  height: 100%;
}

img.FactBannerSmallDimension {
  width: 640px;
}

img.FactBannerMediumDimension {
  width: 720px;
}

img.FactBannerLargeDimension {
  width: 960px;
}
