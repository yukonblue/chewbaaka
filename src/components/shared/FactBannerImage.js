/**
 * FactBannerImage.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Aug 11, 2020
 * Updated  : Aug 23, 2021
 */

import React from 'react'

import './FactBannerImage.css'

export default function FactBannerImage({ src, alt, large, centered, width, height }) {
  let classNames = ["FactBannerImage", "FullWidth", "DisplayBlock", "VerticalCushionPadding"];

  if ( centered ) {
    classNames.push("Centered");
  }

  const sizeClass = large ? "FactBannerMediumDimension" : "FactBannerSmallDimension";

  classNames.push(sizeClass);

  const aspectRatio = {
    width: parseInt(width, 10),
    height: parseInt(height, 10)
  }

  return (
    <img
      className={classNames.join(" ")}
      src={src}
      alt={alt}
      // Specify the aspect ratio
      width={aspectRatio.width}
      height={aspectRatio.height}
    />
  );
}
