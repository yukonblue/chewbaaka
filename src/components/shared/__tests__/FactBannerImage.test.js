/**
 * FactBannerImage.test.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Aug 17, 2020
 * Updated  : Aug 23, 2021
 */

import React from 'react'

import { render } from '@testing-library/react'

import renderer from 'react-test-renderer'

import FactBannerImage from '../FactBannerImage'

const href = "https://google.com/image.jpg";
const src = "Some image";
const width = 520;
const height = 320;

test('renders FactBannerImage component', () => {
  render(
    <FactBannerImage
      src={src}
      href={href}
      width={width}
      height={height}
      large
      centered
    />
  );
});

test('FactBannerImage component snapshot', () => {
  const tree = renderer
    .create(
      <FactBannerImage
        src={src}
        href={href}
        width={width}
        height={height}
        large
        centered
      />
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
