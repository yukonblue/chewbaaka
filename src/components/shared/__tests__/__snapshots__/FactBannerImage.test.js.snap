// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FactBannerImage component snapshot 1`] = `
<img
  className="FactBannerImage FullWidth DisplayBlock VerticalCushionPadding Centered FactBannerMediumDimension"
  height={320}
  src="Some image"
  width={520}
/>
`;
